### ISSUE TITLE
A bug of neural-compressor 2.0

### ISSUE THREAD

**Issue by MeouSker77** [intel/neural-compressor#617]
The following code creates a dataset which has three features (each one is a Tensor with shape (2,)) and one label (a Tensor with shape (1,))
```python
import tensorflow as tf

a = tf.random.normal((1, 2))
b = tf.random.normal((1, 2))
c = tf.random.normal((1, 2))

x= tf.data.Dataset.from_tensor_slices((a, b, c)) # `next(iter(x))` will return (Tensor(2,), Tensor(2,), Tensor(2,))
y = tf.data.Dataset.from_tensor_slices([1]) # `next(iter(y))` will return Tensor(1,)
dataset = tf.data.Dataset.zip((x,y)) # `next(iter(dataset))` will return ((Tensor(2,), Tensor(2,), Tensor(2,)), Tensor(1,))
```

however
```python
from neural_compressor.data import DataLoader
dataloader = DataLoader("tensorflow", dataset)
for e in dataloader:
    print(e)
```
will print something like:
```
[(<generator object TFDataDataLoader._generate_dataloader.<locals>.<genexpr> at 0x7f0675da3900>,), (1,)]
```

Change this line
https://github.com/intel/neural-compressor/blob/3eceb7fcba0517a43e446d9c6bfa85ffe04e07f2/neural_compressor/experimental/data/dataloaders/tensorflow_dataloader.py#L104

to
```python
samples.append(tuple(iter_input.numpy() for iter_input in iter_inputs))
```
may fix this bug.

**Comment by NeoZhangJianyu** [comment]
@MeouSker77 
Thank your contribution!
I will forward to dev team to check and merge if possible.
